be/src/storage/tablet_manager.cpp

```diff
@@ -142,7 +142,34 @@ Status TabletManager::create_tablet(const TCreateTabletReq& request, std::vector
     int32_t schema_hash = request.tablet_schema.schema_hash;
     LOG(INFO) << "Creating tablet " << tablet_id;
 
-    std::unique_lock wlock(_get_tablets_shard_lock(tablet_id));
+    std::unique_lock wlock(_get_tablets_shard_lock(tablet_id), std::defer_lock);
+    std::unique_lock<std::shared_mutex> base_wlock;
+
+    // If do schema change, both the shard where the source tablet is located and
+    // the shard where the target tablet is located need to be locked.
+    // In order to prevent deadlock, the order of locking needs to be fixed.
+    if (request.__isset.base_tablet_id && request.base_tablet_id > 0) {
+        int shard_idx = _get_tablets_shard_idx(tablet_id);
+        int base_shard_idx = _get_tablets_shard_idx(request.base_tablet_id);
+
+        if (shard_idx == base_shard_idx) {
+            wlock.lock();
+        } else {
+            std::unique_lock tmp_wlock(_get_tablets_shard_lock(request.base_tablet_id), std::defer_lock);
+            base_wlock = std::move(tmp_wlock);
+
+            if (shard_idx < base_shard_idx) {
+                wlock.lock();
+                base_wlock.lock();
+            } else {
+                base_wlock.lock();
+                wlock.lock();
+            }
```

Comment:
Check if this can help:
https://en.cppreference.com/w/cpp/thread/lock

Reply:
std::lock may be use high cpu. scoped_load don't support std::move.